test_register_volumes failure

Following up on my previous question, we cloned the repo locally and ran the tests with the current `main` branch.

The results of `pytest` are:

<details>
<summary>pytest results</summary>

```bash
pytest
======================================================================= test session starts =======================================================================
platform linux -- Python 3.13.5, pytest-8.4.1, pluggy-1.6.0
rootdir: /home/dps/Documents/github/warpfield
configfile: pyproject.toml
collected 8 items                                                                                                                                                 

tests/test_all.py ......F.                                                                                                                                  [100%]

============================================================================ FAILURES =============================================================================
______________________________________________________________________ test_register_volumes ______________________________________________________________________

    @pytest.mark.skipif(not gpu_available, reason="No GPU detected.")
    def test_register_volumes():
        """Test the register_volumes function."""
        fixed = np.random.rand(256, 256, 256).astype("float32")
        moving = np.roll(fixed, shift=5, axis=0).copy()  # Simulate a simple shift
        recipe = warpfield.Recipe.from_yaml("default.yml")
    
        registered, warp_map, _ = warpfield.register_volumes(fixed, moving, recipe, verbose=False)
    
        assert registered.shape == fixed.shape, "Registered volume shape mismatch."
>       assert (
            np.abs(registered[10:-10, 10:-10, 10:-10] - fixed[10:-10, 10:-10, 10:-10]) < 0.2
        ).mean() > 0.9, "Registered volume does not match the fixed volume."
E       AssertionError: Registered volume does not match the fixed volume.
E       assert np.float64(0.2959098635936488) > 0.9
E        +  where np.float64(0.2959098635936488) = <built-in method mean of numpy.ndarray object at 0x7f42f53868b0>()
E        +    where <built-in method mean of numpy.ndarray object at 0x7f42f53868b0> = array([[[0.01926519, 0.40094262, 0.6469546 , ..., 0.8788612 ,\n         0.43651927, 0.44386426],\n        [0.75516343, 0...334, 0.07151991, 0.19169569, ..., 0.71982914,\n         0.7857519 , 0.19170444]]], shape=(236, 236, 236), dtype=float32) < 0.2.mean
E        +      where array([[[0.01926519, 0.40094262, 0.6469546 , ..., 0.8788612 ,\n         0.43651927, 0.44386426],\n        [0.75516343, 0...334, 0.07151991, 0.19169569, ..., 0.71982914,\n         0.7857519 , 0.19170444]]], shape=(236, 236, 236), dtype=float32) = <ufunc 'absolute'>((array([[[0.        , 0.        , 0.        , ..., 0.        ,\n         0.        , 0.        ],\n        [0.        , 0...266, 0.6752459 , 0.36060017, ..., 0.        ,\n         0.        , 0.        ]]], shape=(236, 236, 236), dtype=float32) - array([[[0.01926519, 0.40094262, 0.6469546 , ..., 0.8788612 ,\n         0.43651927, 0.44386426],\n        [0.75516343, 0...932, 0.7467658 , 0.55229586, ..., 0.71982914,\n         0.7857519 , 0.19170444]]], shape=(236, 236, 236), dtype=float32)))
E        +        where <ufunc 'absolute'> = np.abs

tests/test_all.py:87: AssertionError
======================================================================== warnings summary =========================================================================
../../../miniconda3/envs/warpfield/lib/python3.13/site-packages/cupyx/jit/_interface.py:173
  /home/dps/miniconda3/envs/warpfield/lib/python3.13/site-packages/cupyx/jit/_interface.py:173: FutureWarning: cupyx.jit.rawkernel is experimental. The interface can change in the future.
    cupy._util.experimental('cupyx.jit.rawkernel')

tests/test_all.py::test_import_nii
  /home/dps/Documents/github/warpfield/src/warpfield/utils.py:56: UserWarning: The NIfTI loader ignores scale and offset. Please ensure that fixed and moving volumes are at the same scale and orientation.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
===================================================================== short test summary info =====================================================================
FAILED tests/test_all.py::test_register_volumes - AssertionError: Registered volume does not match the fixed volume.
============================================================ 1 failed, 7 passed, 2 warnings in 10.04s =============================================================
```
</details>

Is this failure expected? I just want to make sure we are starting from the right place.

--- a/tests/test_all.py
+++ b/tests/test_all.py
@@ -77,7 +77,10 @@ def test_import_tiff(tmp_path):
 @pytest.mark.skipif(not gpu_available, reason="No GPU detected.")
 def test_register_volumes():
     """Test the register_volumes function."""
+    import cupyx.scipy.ndimage
+
     fixed = np.random.rand(256, 256, 256).astype("float32")
+    fixed = cupyx.scipy.ndimage.gaussian_filter(cp.array(fixed), sigma=1).get()
     moving = np.roll(fixed, shift=5, axis=0).copy()  # Simulate a simple shift
     recipe = warpfield.Recipe.from_yaml("default.yml")
 

